packages/common/src/ceramic-api.ts

```diff
@@ -77,6 +92,12 @@ export interface AdminApi {
    */
   startIndexingModels(modelsIDs: Array<StreamID>): Promise<void>
 
+  /**
+   * Index models, with custom indices
+   *
+   */
+  startModelIndexing(models: Map<StreamID, Array<ModelIndex> | undefined>): Promise<void>
```

Comment:
Well I really hate having both `startIndexingModels` and `startModelIndexing` with slightly different behavior.

I guess one option is to rename the new one something like `startIndexingModelsWithCustomIndexes`, but that's a mouthful, especially since the old version should eventually go away.

What I'd really like to do is rename the old function to `startIndexingModelsV0_deprecated` and call the new one `startIndexingModels`, but then that's a breaking change to any apps that build with the ceramic http client directly, so we'd probably need to do a major version bump.  Of course at that point if we're doing a major version bump we could just remove the old form of the API entirely and replace it wholesale with the new format.

@oed for thoughts on doing a major version bump.  CC @michaelsena

Reply:
Is there no way we could just use the old api by adding an additional parameter? Looks to me like it chould be fairly straight forward to do so by having some extra logic in the [_validateAdminApiJWS](https://github.com/ceramicnetwork/js-ceramic/blob/37c2297b6f32e73543d7f0c06413a2b67c27cd5f/packages/cli/src/ceramic-daemon.ts#L694) method?


Then we just change the function signature of `startIndexingModels` to:

`startIndexingModels(modelsIDs: Array<StreamID>, indexes?: Map<StreamID, Array<ModelIndex>): Promise<void>`

Reply:
Changing the function signature in this way is a little weird, since we either have to make sure the modelIDs match and discard indices that don't, or just allow indexing from either fields.

Reply:
Ok, I think we could do with only having the `startModelIndexing(models: Map<StreamID, Array<ModelIndex> | undefined>): Promise<void>` signature though? And deal with backwards compatibility in the [_validateAdminApiJWS](https://github.com/ceramicnetwork/js-ceramic/blob/37c2297b6f32e73543d7f0c06413a2b67c27cd5f/packages/cli/src/ceramic-daemon.ts#L694) method?

Reply:
I think that's breaking for anyone using startModelIndexing. For now, just going to assume that you can index models from either field, joining those arrays together.